README.md

```diff
@@ -31,6 +31,101 @@ Application-level tracing for Rust.
 structured, event-based diagnostic information. `tracing` is maintained by the
 Tokio project, but does _not_ require the `tokio` runtime to be used.
 
+## Usage
+
+### In libraries
+
+Libraries should only rely on the `tracing` crate and use the provided macros
+and types to collect whatever information might be useful to downstream consumers.
+
+```toml
+[dependencies]
+tracing = "0.1"
+```
+
+```rust
+use tracing::{debug, error, info, span, trace, warn, Level};
+
+use std::{error::Error, fmt};
+
+#[tracing::instrument]
+pub fn shave(yak: usize) -> Result<(), Box<dyn Error + 'static>> {
+    debug!(
+        message = "hello! I'm gonna shave a yak.",
+        excitement = "yay!"
+    );
+    if yak == 3 {
+        warn!(target: "yak_events", "could not locate yak!");
+        return Err(ShaveError::new(yak, YakError::new("could not locate yak")).into());
+    } else {
+        trace!(target: "yak_events", "yak shaved successfully");
+    }
+    Ok(())
+}
+
+pub fn shave_all(yaks: usize) -> usize {
+    let span = span!(Level::TRACE, "shaving_yaks", yaks_to_shave = yaks);
+    let _enter = span.enter();
+
+    info!("shaving yaks");
+
+    let mut num_shaved = 0;
+    for yak in 1..=yaks {
+        let res = shave(yak);
+        trace!(target: "yak_events", yak, shaved = res.is_ok());
+
+        if let Err(ref error) = res {
+            error!(
+                message = "failed to shave yak!",
+                yak,
+                error = error.as_ref()
+            );
+        } else {
+            num_shaved += 1;
+        }
+
+        trace!(target: "yak_events", yaks_shaved = num_shaved);
+    }
+
+    num_shaved
+}
+```
+
+### In applications
+
+In order to produce output, applications need a concrete subscriber implementation. 
+[`tracing_subscriber`](https://docs.rs/tracing-subscriber/) is a reasonable default.
+By default, `tracing-subscriber` is able to consume messages emitted by
+`log`-instrumented libraries and modules.
+
+```toml
+[dependencies]
+tracing = "0.1"
+tracing-subscriber = "0.2.0-alpha.2"
+```
+
+```rust
+use tracing::debug;
+use tracing_subscriber;
+use yak_shave;
+
+fn main() {
+    tracing_subscriber::fmt::init();
+
+    let number_of_yaks = 3;
+    debug!("preparing to shave {} yaks", number_of_yaks);
+
+    let number_shaved = yak_shave::shave_all(number_of_yaks);
+
+    debug!(
+        message = "yak shaving completed.",
+        all_yaks_shaved = number_shaved == number_of_yaks,
+    );
+}
+```
+
+Note that if you are instrumenting libraries or applications that make use of [`std::future::Future`](https://doc.rust-lang.org/stable/std/future/trait.Future.html) or async/await, be sure to use the [`tracing-futures`](https://docs.rs/tracing-futures) crate.
```

Comment:
This is a little unhelpful, IMHO. Maybe instead of "be sure to use", we should actually explain _how_ to use it...

Reply:
Good point. I didn't get at this one sorted yet, but I'd like to have a "usage notes" section under the "in libraries" and "in applications" section, and this + an example will start that off. I'll update this PR to include that.

Reply:
added a usage notes + async code example.